apps/sr/tid1500reader.cxx

```diff
@@ -0,0 +1,218 @@
+  st.getCursorToRootNode(cursor);
+  if(st.gotoNamedChildNode(CODE_DCM_ImagingMeasurements)) {
+    // can have 1-n MeasurementGroups
+    do {
+      if (st.gotoNamedChildNode(CODE_DCM_MeasurementGroup)) {
+        Json::Value measurement;
+        if (st.gotoNamedChildNode(DSRCodedEntryValue("C67447", "NCIt", "Activity Session"))) {
+          // TODO: think about it
+          cout << "Activity Session: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["ActivitySession"] = st.getCurrentContentItem().getStringValue().c_str();
+        }
+        st.gotoParent();
+        if (st.gotoNamedChildNode(CODE_DCM_ReferencedSegment)) {
+          DSRImageReferenceValue referenceImage = st.getCurrentContentItem().getImageReference();
+          OFVector<Uint16> items;
+          referenceImage.getSegmentList().getItems(items);
+          cout << "Reference Segment: " << items[0] << endl;
+          measurement["ReferencedSegment"] = items[0];
+          if (!referenceImage.getSOPInstanceUID().empty()){
+            measurement["segmentationSOPInstanceUID"] = referenceImage.getSOPInstanceUID().c_str();
+          }
+          st.gotoParent();
+        }
+        st.gotoParent();
+        if (st.gotoNamedChildNode(CODE_DCM_SourceSeriesForSegmentation)) {
+          cout << "SourceSeriesForImageSegmentation: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["SourceSeriesForImageSegmentation"] = st.getCurrentContentItem().getStringValue().c_str();
+          st.gotoParent();
+        }
+        if (st.gotoNamedChildNode(CODE_DCM_TrackingIdentifier)) {
+          cout << "TrackingIdentifier: " << st.getCurrentContentItem().getStringValue().c_str() << endl;
+          measurement["TrackingIdentifier"] = st.getCurrentContentItem().getStringValue().c_str();
+          st.gotoParent();
+        }
+        if (st.gotoNamedChildNode(CODE_DCM_Finding)) {
+          measurement["Finding"]["CodeMeaning"] = st.getCurrentContentItem().getCodeValue().getCodeMeaning().c_str();
+          measurement["Finding"]["CodingSchemeDesignator"] = st.getCurrentContentItem().getCodeValue().getCodingSchemeDesignator().c_str();
+          measurement["Finding"]["CodeValue"] = st.getCurrentContentItem().getCodeValue().getCodeValue().c_str();
+        }
+        st.gotoParent();
+        if (st.gotoNamedChildNode(DSRCodedEntryValue("G-C0E3", "SRT", "Finding Site"))) {
+          measurement["FindingSite"]["CodeMeaning"] = st.getCurrentContentItem().getCodeValue().getCodeMeaning().c_str();
+          measurement["FindingSite"]["CodingSchemeDesignator"] = st.getCurrentContentItem().getCodeValue().getCodingSchemeDesignator().c_str();
+          measurement["FindingSite"]["CodeValue"] = st.getCurrentContentItem().getCodeValue().getCodeValue().c_str();
+        }
+        st.gotoParent();
+        st.gotoChild();
+
+        Json::Value measurementItems(Json::arrayValue);
+        while (st.gotoNext()){
```

Comment:
Yes, that's a good test. Having problems if there is more than 1 MeasurementGroup. Need to check that....